src/templates/template-common/utils.py

```diff
@@ -58,7 +58,21 @@ def setup_config(config_path, backend, **kwargs):
     return config
 
 
-#::: } else { :::#
+#::: } else if ((it.argparser == 'hydra')) { :::#
+
+
+def setup_config(config):
+    optional_attributes = ["output_dir", "backend", "num_iters_per_epoch"]
+
+    for attr in optional_attributes:
+        if attr == "output_dir":
+            OmegaConf.update(config, attr, Path(config.get(attr, "./")), force_add=True)
+        if config.get(attr, None) == None:
+            OmegaConf.update(config, attr, None, force_add=True)
```

Comment:
can you explain this code, I'm not sure to understand this part why we need this.

Reply:
Since hydra behaves a certain way in the running of experiment, I had problems adding new parameters in  `DictConfig` during running of the experiment, especially when we use spawn processes like [here](https://github.com/pytorch-ignite/code-generator/actions/runs/5963864652/job/16177913613) as it seems that it doesn't behave well with idist process with different ranks, so I have added optional parameters as defaults if they are not present in `config.yaml`. Not very sure how to proceed with this

Reply:
The problem under link is with idist.broadcast on PosixPath: https://github.com/pytorch-ignite/code-generator/actions/runs/5963864652/job/16177913613#step:17:53

For me this looks like a random addition. Try to figure out and explain in a clear way what happens and why we need to add this code.

Reply:
I think the reason is that `@hydra.main()` seems to compose the `config` initially and so if we want to add any new variable, we have t add it using `OmegaConf.update` method or using [`hydra compose API`](https://hydra.cc/docs/advanced/compose_api/). Lets add the optional parameters using force update method in `setup_config` method  as done before and mention that if you want to add any new parameter in the config during any part of the run function, you have to add the parameter in the setup_config as well. This seems more organised.